storage/filesystem/object.go

```diff
@@ -268,7 +269,9 @@ func (s *ObjectStorage) getFromPackfile(h plumbing.Hash, canBeDelta bool) (
 		return nil, err
 	}
 
-	defer ioutil.CheckClose(f, &err)
+	if !s.dir.Options.KeepDescriptors {
```

Comment:
Instead of requesting the options to dogGit, should be accessed from a local pointer to the storer.